README.md

```diff
@@ -2,8 +2,147 @@
+do not transfer file contents. The service just passes data about file location. The file content has to be provided by other service (eg. upscan)
+
+# Typical usage scenario
+
+- Consuming service requests upload of the file from user using `upscan` service
+- `Upscan` notifies consuming service that the file has been uploaded and is ready to be processed - giving consuming 
+service the URL that allows to download the file.
+- Consuming service ensures that all required files have been uploaded by the user
+- Consuming service uses file-transmission service to notify MDG that files are ready to be processes
+- MDG asynchronously uses the file
+
+## Service usage
+
+### Whitelisting
+
+In order to initiate an upload the consuming service must be whitelisted by file-transmission. See the 'Whitelisting client services' section further down in this document.
+
+### Requesting a file transmission
+
+The basic unit of work for file-transmission service is a batch that might consist of one or more files.
+
+Information about every file in the batch is passed to the file-transmission service as a separate POST request. 
+There is no need to make additional call to create a batch or to notify that data about all files in the batch have 
+been provided.
+
+Assuming the consuming service is whitelisted, it makes a POST request to the `/file-transmission/request` endpoint. 
+The request contains data about batch, details of the file and any additional metadata which consuming service wants
+to pass to MDG. The request should also contain callback URL that will be used to asynchronously notify consuming service 
+that MDG has processed the request.
+
+The following information has to pe passed within request:
+- Batch identifier
+- Number of files in the batch
+- `callbackUrl` that will be used by MDG to notify upstream service about result of batch processing.
+- `journeyName` / `journeyVersion` - information for MDG about meaning of files in the batch, determining what process should be invoked.
+- `requestTimeoutInSeconds` - time describing how long file-transmission service will try to deliver file details to MDG. 
+- Details about the file
+  - `location` - URL allowing to download file contents. Please be aware that this URL has to be accessible by MDG (verify networking configuration and use external domain names). Download URL's provided by upscan service are valid.
+  - `fileName` / `mimeType` - original name of the file and it's MIME type
+  - `checksum` - SHA256 checksum of the file in hexadecimal format
+  - `reference` - unique reference of the file (this will be interpreted by MDG as `correlationId`)
+  - `sequenceNumber` -  sequence number of the file in the batch (**please be aware that this should be 1-based**)
+- Additional properties - key/value map of custom properties that can be attached to the file
+
+The request has to include the following HTTP headers:
```

Comment:
If some of the headers need not be required, by definition, they do not 'have to be included'.